datadog/resource_datadog_service_level_objective.go

```diff
@@ -242,6 +383,8 @@ func buildServiceLevelObjectiveStructs(d *schema.ResourceData) (*datadogV1.Servi
 			}
 			slo.SetMonitorIds(s)
 			slor.SetMonitorIds(s)
+		} else {
+			return nil, nil, fmt.Errorf("monitor_ids is required for monitor SLOs")
```

Comment:
New checks that ensure useful error messages when logically required fields are missing.